### DISCUSSION TITLE
why not provide detection score in the result?

### DISCUSSION THREAD

**Discussion by MengHao666** [PaddlePaddle/PaddleOCR#14453] [Q&A]
Now the script only provide recognition score.

**Comment by GreatV** [comment]
The current behavior of the script in PaddleOCR is to provide only the **recognition confidence score** in the inference results. This means the confidence score reflects the model's certainty regarding the text recognition step, but not the text **detection** step.

If you're wondering why the **detection confidence score** isn't provided, it may be due to the following reasons:

1. **Implementation Focus**: The primary focus of many OCR systems, including PaddleOCR, is on providing accurate recognition results. Therefore, more emphasis is placed on the recognition confidence score rather than detection.

2. **Detection Confidence Usage**: While the detection stage does involve confidence metrics (e.g., bounding box probabilities), these scores might not be exposed by default in the output, as they are often used internally for filtering or processing the detected regions.

3. **Accessing Detection Scores**: If you require detection confidence scores, it might involve modifying the code. Based on the discussions in the PaddleOCR community (e.g., [Discussion #14009](https://github.com/PaddlePaddle/PaddleOCR/discussions/14009)), users have inquired about this feature. However, as of now, the PaddleOCR framework does not seem to provide detection confidence scores directly in the results.

### Possible Workarounds:
- **Code Modification**: You may need to modify the detection module in the PaddleOCR codebase to extract and include the detection confidence scores in the output.
- **Community Input**: Engage with the GitHub discussion (e.g., [Discussion #14009](https://github.com/PaddlePaddle/PaddleOCR/discussions/14009)) to suggest this as a feature request or get guidance on how to implement it yourself.

If this feature is critical for your application, customizing the PaddleOCR code or monitoring future updates may be necessary.

---
*Response generated by [feifei-bot](https://github.com/GreatV/feifei) | chatgpt-4o-latest*
